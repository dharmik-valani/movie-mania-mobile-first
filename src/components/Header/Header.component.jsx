import React from "react";
import MoreIcon from "../../assets/Header/moreicon.svg";
import backicon from "../../assets/Header/backicon.svg";
import Imagecomponent from "../Image/Image.component";

const Header = ({ text, backbutton, morebutton ,onClick}) => {


  return (
    <div className="bg-[#212121] p-4 sticky top-0 z-10" data-testid={text}>
      <div className="flex justify-between">
        <div className="flex">
          {backbutton && (
            <Imagecomponent
              src={backicon}
              alttext={"back-arrow"}
              imageclassName={"mr-2 cursor-pointer"}
              onClick={onClick}
            />
          )}
          <div className="text-white text-xl">{text}</div>
        </div>
        {morebutton && (
          <Imagecomponent src={MoreIcon} alttext={"more-icon"} imageclassName={"cursor-pointer"} />
        )}
      </div>
    </div>
  );
};

export default Header;
