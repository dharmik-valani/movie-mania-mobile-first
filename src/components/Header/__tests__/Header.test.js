import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Header from "../Header.component";

describe("Header", () => {
  let originalFetch;

  beforeEach(() => {
    originalFetch = global.fetch;
    global.fetch = jest.fn(() =>
      Promise.resolve({
        json: () =>
          Promise.resolve([
            {
              text: "Pop Movies",
              morebutton: true,
            },
          ]),
      })
    );
  });

  afterEach(() => {
    global.fetch = originalFetch;
  });

  it("Renders Header component", async () => {
    let propData = {
      text: "Pop Movies",
      morebutton: true,
    };

    render(<Header {...propData} />);
    const linkElement = await screen.getByTestId("Pop Movies");
    expect(linkElement).toBeInTheDocument();
  });

  //  SNAP SHOT
  test("matches header snapshot", async () => {
    let propData = {
      text: "Pop Movies",
      morebutton: true,
    };
    const tree = await renderer.create(<Header {...propData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
