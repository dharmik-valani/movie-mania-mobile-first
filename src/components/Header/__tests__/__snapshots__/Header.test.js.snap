// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches header snapshot 1`] = `
<div
  className="bg-[#212121] p-4 sticky top-0 z-10"
  data-testid="Pop Movies"
>
  <div
    className="flex justify-between"
  >
    <div
      className="flex"
    >
      <div
        className="text-white text-xl"
      >
        Pop Movies
      </div>
    </div>
    <img
      alt="more-icon"
      className="cursor-pointer"
    />
  </div>
</div>
`;
